Update ci-cd docs (#933)
kyma-project/api-gateway: docs/contributor/04-30-ci-cd.md

Reviewer: We are missing the [UI-Tests workflow](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/ui-tests.yaml#L23).

Reviewer: Do we need to have the [release workflow](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/create-release.yaml) also documented? Since this is also about the CD flow, I think it should be included.

| [`PR Integration Release / Integration tests`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/pull-integration-release.yaml#L37)                     | Executes the main integration test suite for the API Gateway module on a k3d cluster. The job installs the Istio module using Kyma CLI.                                    |
| [`PR Integration Release / Kubernetes version compatibility test`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/pull-integration-release.yaml#L54) | Executes the main integration test suite for the API Gateway module and checks for any incompatibilities with the latest Kubernetes version.                               |
| [`PR Integration Release / Upgrade tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/pull-integration-release.yaml#L71)          | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version.               |

> | [`Pull Request / Golang lint check (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-request.yaml#L24)            | Performs linting.                                                                                                                                                          |

Reviewer: We might also want to use permalinks for `Pull Request / Golang lint check (pull_request)`, `Pull Request / Run unit tests (pull_request)` and `pre-api-gateway-presubmit-scanner`.

| [`PR Integration Release / Custom domain GCP`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/pull-integration-release.yaml#L88)                                                                                                                                           | Executes the integration test suite that verifies the functional correctness of Kyma API Gateway Operator. The test suite uses a custom domain and a Gardener GCP cluster. |
| [`PR Integration Release / Custom domain AWS`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/pull-integration-release.yaml#L107)                                                                                                                                           | Executes the integration test suite that verifies the functional correctness of the API Gateway Operator. The test suite uses a custom domain and a Gardener AWS cluster.  |

> ## Pipelines Running on the Main Branch

Reviewer 2: ```suggestion
## Pipelines Running on the Main Branch and on a Schedule

The following CI jobs are executed when you merge changes to the main branch and also run daily at 5:00 AM.
```

| [`Main Integration / Custom domain integration GCP`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/main-integration.yaml#L75)                           | Executes the integration test suite that verifies the functional correctness of Kyma API Gateway Operator. The test suite uses a custom domain and a Gardener GCP cluster. |
| [`Main Integration / Custom domain integration AWS`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/main-integration.yaml#L94) | Executes the integration test suite that verifies the functional correctness of the API Gateway Operator. The test suite uses a custom domain and a Gardener AWS cluster.  |
| [`Main Integration / Slack Notification`](https://github.com/kyma-project/api-gateway/blob/3d10f7c7801f514524a267239924fdc889f2be3f/.github/workflows/main-integration.yaml#L113)                                      | Sends a Slack notification to the team's channel if any pipelines fail on the main branch.                                                                                 |

## Pipelines Running on the Release Branch

> The following CI jobs generate release artifacts. They are triggered on Git tag creation.

Reviewer 2: ```suggestion
The following CI jobs generate release artifacts. They are triggered when you create a Git tag.
```

> The following CI jobs generate release artifacts. They are triggered on Git tag creation.

Reviewer: Do we need to have the [release workflow](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/create-release.yaml) also documented? Since this is also about the CD flow, I think it should be included.

| Name                                                                                                                                              | Description                                                                                 |
|---------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------|
| [`rel-api-gateway-manager-build`](https://github.com/kyma-project/test-infra/blob/6fa2829ff262d300663335f80f67050d21b1ae5f/prow/jobs/kyma-project/api-gateway/api-gateway-manager-build.yaml#L112) | Builds Kyma API Gateway Operator image on the release and pushes it to the `prod` registry. |

## Pipelines Running on a schedule

> Pipelines that run on shedule are the same as those running on merge to `main` branch. Those run every day at 05:00 AM.

Reviewer: Typo
```suggestion
Pipelines that run on schedule are the same as those running on merge to `main` branch. Those run every day at 05:00 AM.
```
